azurerm/resource_arm_virtual_machine_scale_set_test.go

```diff
@@ -85,6 +85,26 @@ func TestAccAzureRMVirtualMachineScaleSet_basicPublicIP(t *testing.T) {
 	})
 }
 
+func TestAccAzureRMVirtualMachineScaleSet_basicApplicationSecurity(t *testing.T) {
+	resourceName := "azurerm_virtual_machine_scale_set.test"
+	ri := acctest.RandInt()
+	config := testAccAzureRMVirtualMachineScaleSet_basicApplicationSecurity(ri, testLocation())
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testCheckAzureRMVirtualMachineScaleSetDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: config,
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMVirtualMachineScaleSetExists(resourceName),
+					testCheckAzureRMVirtualMachineScaleSetApplicationSecurity(resourceName),
```

Comment:
rather than reaching out and checking the state of the resource in Azure - we should be able to verify this in the local state by switching this to using a generic test check function like so:

```
resource.TestCheckResourceAttr(resourceName, "ip_configuration.0.application_security_group_ids.#", "1")
```

Reply:
Ok, I will update this test.